Library/Homebrew/rubocops/components_order.rb

```diff
@@ -130,6 +195,56 @@ def get_state(node1)
             return [idx, comp.index(node1), comp] if comp.member?(node1)
           end
         end
+
+        def check_order(component_precedence_list, body_node)
+          present_components = component_precedence_list.map do |components|
+            components.flat_map do |component|
+              case component[:type]
+              when :method_call
+                find_method_calls_by_name(body_node, component[:name]).to_a
+              when :block_call
+                find_blocks(body_node, component[:name]).to_a
+              when :method_definition
+                find_method_def(body_node, component[:name])
+              end
+            end.compact
+          end
+
+          # Check if each present_components is above rest of the present_components
+          offensive_nodes = nil
+          present_components.take(present_components.size - 1).each_with_index do |preceding_component, p_idx|
+            next if preceding_component.empty?
+
+            present_components.drop(p_idx + 1).each do |succeeding_component|
+              next if succeeding_component.empty?
+
+              offensive_nodes = check_precedence(preceding_component, succeeding_component)
+              break if offensive_nodes
+            end
+          end
+
+          [present_components, offensive_nodes]
+        end
+
+        # Method to format message for reporting component precedence violations
+        def component_problem(c1, c2)
+          # `aspell`: options and resources should be grouped by language
+          whitelist = %w[
+            aspell
+          ].freeze
```

Comment:
```suggestion
          whitelist = %w[
            aspell
          ].freeze
```

Make this a constant `COMPONENT_WHITELIST` or similar and add to `test/rubocops/components_spec.rb`. See the `include_examples "formulae exist", described_class::MAKE_CHECK_WHITELIST` for an example.